core/embed/extmod/modtrezorio/modtrezorio-fatfs.h

```diff
@@ -188,6 +188,10 @@ STATIC mp_obj_t mod_trezorio_FatFSFile_write(mp_obj_t self, mp_obj_t data) {
   if (res != FR_OK) {
     mp_raise_OSError(fresult_to_errno_table[res]);
   }
+  if (written != buf.len) {
```

Comment:
Will this ever be reached? I mean if less than `buf.len` bytes are written would the `f_write` above return `FR_OK`?

Reply:
If you write more than available space on the filesystem or when number of free clusters falls to 0, the write will still return `FR_OK`, but the `written` will contain actual number of bytes written.